cinder/volume/manager.py

```diff
@@ -680,6 +680,36 @@ def _set_resource_host(self, resource):
             resource.host = volume_utils.append_host(self.host, pool)
             resource.save()
 
+    def _propagate_volume_scheduler_hints(self, context, volume):
+        """Ensure metadata hints are propagated to referenced volumes."""
+        hints = volume_utils.get_scheduler_hints_from_volume(volume)
+        if not hints:
+            return
+
+        LOG.debug("Found hints for %(volume)s - %(hints)s",
+                  {'volume': volume.id, 'hints': hints})
+        for hint_key in hints:
+            for vol_id in hints[hint_key]:
+                try:
+                    meta_vol = objects.Volume.get_by_id(context, vol_id)
+                    # Because pep8 error otherwise with lines 696 and 710
+                    vut = volume_utils
+                    meta_vol_hints = vut.get_scheduler_hints_from_volume(
+                        meta_vol
+                    )
+                    meta_vol_hints.setdefault(hint_key, [])
+                    if volume.id not in meta_vol_hints[hint_key]:
+                        meta_vol_hints[hint_key].append(volume.id)
+
+                    md = vut.set_scheduler_hints_to_volume_metadata(
+                        meta_vol_hints, meta_vol.metadata
+                    )
+                    meta_vol.metadata = md
+                    meta_vol.save()
+                except Exception:
+                    LOG.error("Failed to set scheduler hints",
+                              resource=meta_vol)
```

Comment:
I think it would be better to use `LOG.exception()` here, so we can debug the problem properly.

Reply:
done